[Term Entry] Dart Type Conversion .toStringAsExponential() (#4699)
Codecademy/docs: content/dart/concepts/type-conversion/terms/toStringAsExponential/toStringAsExponential.md

> ---
> Title: '.toStringAsExponential()'
> Description: 'Converts a number to its exponential notation in a string format.'
> Subjects:
>   - 'Computer Science'
>   - 'Mobile Development'
>   - 'Code Foundations'
> Tags:
>   - 'Dart'
>   - 'String'
>   - 'Method'
> CatalogContent:
>   - 'learn-dart'
>   - 'paths/computer-science'
> ---

Reviewer: ```suggestion
---
Title: '.toStringAsExponential()'
Description: 'Converts a number to its exponential notation in a string format.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
  - 'Mobile Development'
Tags:
  - 'Dart'
  - 'String'
  - 'Method'
CatalogContent:
  - 'learn-dart'
  - 'paths/computer-science'
---
```

> The **`.toStringAsExponential()`** method is a Dart type conversion method that first converts a number to its exponential notation, then changes it to a string format. It can be used on integers and doubles.

Reviewer: ```suggestion
The **`.toStringAsExponential()`** method in Dart is used for type conversion, converting a number to exponential notation and then converting it to a string format. It can be applied to both integers and doubles.
```

## Syntax

```pseudo
number.toStringAsExponential([int fractionDigits])
```

> - `number`: An integer or double
> - `fractionDigits`: An integer ranging from 0 to 20 that appears after the decimal point. If not specified, the default is as many digits as necessary.

Reviewer: ```suggestion
- `number`: This refers to a numeric value, either an integer or a double, on which the `.toStringAsExponential()` method is called.
- `fractionDigits`: An integer from 0 to 20 indicating the digits after the decimal point. If not specified, it defaults to as many digits as needed for accuracy.
```

## Example

The example below shows how `.toStringAsExponential()` method is used:

```dart
void main() {
  double number = 23625.585;
  String result1 = number.toStringAsExponential();
  String result2 = number.toStringAsExponential(5);
  print("Without fraction digits: ${result1}");
  print("With fraction digits: ${result2}");
}
```

> `toStringAsExponential(5)` specifies that we want 5 digits after the decimal point.

Reviewer: ```suggestion
This Dart example uses `.toStringAsExponential()` on a double to show both default and customized exponential notation with 5 fraction digits specified. This generates the following output:
```
